MedTracker/Controllers/DoctorController.cs

```diff
@@ -110,6 +109,83 @@ public async Task<IActionResult> GetTreatments()
             return StatusCode(500, $"An error occurred:{ex.Message}");
         }
     }
+    [HttpGet("GetPatient")]
+    [Authorize]
+    public async Task<IActionResult> GetPateint()
```

Comment:
Lets look just at the code, not at the task you should do.
We see method `GetPatient`. We understand this is singular form and an id should be given as argument, but I don't see one.

This method returns a list of `treatmentsWithPatients`, here becomes even more confusing. Are we receiving a patient, a list of patients, a list of treatments or a list of treatments with patients?
Based on the code we can't tell what are the intentions.

Reply:
Now it is GetPatients and it return a list of treatments with patients. In example wroted by you upper is method is GetTreatments and it return the list of treatments with medications. How would you like to name this method?

Reply:
> Now it is GetPatients and it return a list of treatments with patients

If it was at least `patients WITH treatments`, but you are returning `treatments WITH patients`.

> In example wroted by you upper is method is GetTreatments

Yes, there is already a method that returns treatments. Why to make another one?

> In example wroted by you upper is method is GetTreatments and it return the list of treatments with medications

A patient without a treatment is not a patient anymore, is simply a "healthy" human. But what is a treatment without medications?
Ideally yes, if they are treated as two entities, there could have been two methods, one for treatment, as it is in database, and another just for medications. Thanks for pointing me out on this, I made that at night in hurry :) I think I'll split that in another PR.

> How would you like to name this method?

Not renaming, but returning just the patients.

Reply:
I will try to return patients with treatments because in another way how would web know which treatments have which patients

Reply:
Now it returns patient with treatments

Reply:
> Now it returns patients with treatments

But I asked to return just the patients in this method.

> how would we know which treatments have which patients

By creating another method "GetPatientTreatments", which will receive the PatientId.

Reply:
I added new method GetPatientTreatments which return id of a patient and treatment and modified method GetPatient to return information about patient